### DISCUSSION TITLE
Python SQLite3 Database

### DISCUSSION THREAD

**Discussion by jag-k** [vercel/vercel#4561] [Help] [answered]
Hello! I have a question. How can I use SQLite3 DataBase on [Vercel](https://vercel.com)? If is impossible, which database can I use?

> I use Python + [PonyORM](http://ponyorm.org/) + [Sanic](https://sanicframework.org/)

P.S.: Can I upload images/files to server?

**Comment by coetry** [comment, marked as answer]
Hi @jag-k ! @simonw has done some work with SQLite databases on Vercel. Check out his [datasette](https://github.com/simonw/datasette-publish-now) project and see if it meets your needs.

You should keep in mind that Vercel is a frontend first solution and Serverless Functions are meant to server as lightweight API helpers. They have constraints like a stateless filesystem, so if you need long-term persistence, you should consider a third-party storage solution.

**Reply by jag-k** [reply]
Thank you very much!)
